hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -274,19 +275,27 @@ private boolean isBlockCorrupt(int blocksize) throws IOException {
   }
 
   private long scanForNextAvailableBlockOffset() throws IOException {
+    // Make buffer large enough to scan through the file as quick as possible especially if it is on S3/GCS.
+    // Using lower buffer is incurring a lot of API calls thus drastically increasing the cost of the storage
+    // and also may take days to complete scanning trough the large files.
+    byte[] dataBuf = new byte[1024 * 1024];
```

Comment:
Instead of this, can we do the following in the constructor ?

    if (fsDataInputStream.getWrappedStream() instanceof FSInputStream **|| fsDataInputStream.getWrappedStream() instanceof FSDataInputStream**) {
      this.inputStream = new TimedFSDataInputStream(logFile.getPath(), new FSDataInputStream(
          new BufferedFSInputStream((FSInputStream) fsDataInputStream.getWrappedStream(), bufferSize)));
    }

Reply:
After this change, we can leave the scanForNextAvailableBlockOffset() unchanged ? Also, please add comments as to why we need to add the extra check to make the inputStream Buffered

Reply:
I would consider keeping a new one as it does a lot less of memory copies and a lot less additional overhead, technically it is more efficient from the resource utilization stand point.

Reply:
@vburenin Just to confirm, by "lot less additional overhead" you refer to the in-memory to in-memory bytes copy operation that needs to be done for every 6 bytes vs 1MB in a single go in this implementation ? (Since the comparison of byte arrays is the same). Can we quantify this overhead ? 
Additionally, what is the reasoning behind keeping it at 1MB ?

Reply:
Buffered reader needs to check a few things to copy the right data, readFully logic itself is not trivial, there is also position modification each time it reads 6 bytes, etc so even without profiling it I bet the overhead is significant.
1MB seems like a good number to me, not too much, not too little. From my past experience dealing with FS IO going with blocks larger than 1MB was giving a diminished return. However, the best number would be the one that matches underlying block read size, but that depends on the reader which can be any.

Reply:
@vburenin According to the current code, we still are using BufferedReader for all cases except GCS, so that doesn't go away with this code in a generic way. Additionally, we need buffered reader code (the one I pointed above) anyways for GCS given the happy code path (without the need to find the magic header in corrupt blocks) since this method is only called when it encounters a corrupt block. 
1) readFully does a bunch of if conditions so branching could cause some perf degradation here, don't see any other extra logic apart from copying bytes which is the same for irrespective of doing 6 bytes vs 1MB
2) Position modification in BufferedInputStream is a variable assignment which should not cause any overhead. 

Agree with you that the best number should be the one that matches the underlying block size. It would be great if you can do some microbenchmarking here. I'm OK to land this once you can add the if check for BufferedInputStream since that is needed anyways ?

Reply:
All that simple logic runs every time for every byte offset, it all adds up no matter what, I also forgot to mention that we iterate over the data changing position just for 1 byte to copy next 6 bytes every time, so that it can potentially copy 6 times more data, which is not ideal.

I will add a BufferedInputStream as soon as I get back to work on Tuesday. It is super hard to find even 5 minutes when all kids are home.